assets/javascripts/discourse/components/channel-list.js

```diff
@@ -52,6 +54,40 @@ export default Component.extend({
     }`;
   },
 
+  @action
+  browseChannels() {
+    this.router.transitionTo("chat.browse");
+    return false;
+  },
+
+  @action
+  pencilClicked() {
+    if (this.currentUser.staff) {
+      this.togglePopupMenu();
+    } else {
+      this.browseChannels();
+    }
+    return false;
+  },
+
+  @bind
+  togglePopupMenu() {
+    this.set("showPopup", !this.showPopup);
+    next(() => {
+      if (this.showPopup) {
+        window.addEventListener("click", this.togglePopupMenu);
+      } else {
+        window.removeEventListener("click", this.togglePopupMenu);
+      }
+    });
+  },
```

Comment:
imo select-kit would be better here than an ad-hoc menu solution

Reply:
I don't think I agree - the pencil button opens the popup,  I don't even know how this can be done with select-kit